tests/Unit/ApparentHorizons/Test_ApparentHorizonFinder.cpp

```diff
@@ -0,0 +1,350 @@
+    // to speed up the test.
+    Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
+    CHECK_ITERABLE_CUSTOM_APPROX(horizon_radius, get(expected_radius),
+                                 custom_approx);
+    ++test_kerr_horizon_called;
+  }
+};
+
+template <typename Metavariables, typename InterpolationTargetTag>
+struct mock_interpolation_target {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using action_list = tmpl::list<>;
+  using component_being_mocked =
+      intrp::InterpolationTarget<Metavariables, InterpolationTargetTag>;
+  using initial_databox = db::compute_databox_type<
+      typename intrp::Actions::InitializeInterpolationTarget<
+          InterpolationTargetTag>::template return_tag_list<Metavariables>>;
+
+  using const_global_cache_tag_list = Parallel::get_const_global_cache_tags<
+      tmpl::list<typename InterpolationTargetTag::compute_target_points,
+                 typename InterpolationTargetTag::post_interpolation_callback>>;
+};
+
+template <typename Metavariables>
+struct mock_interpolator {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<>;
+  using component_being_mocked = intrp::Interpolator<Metavariables>;
+  using initial_databox =
+      db::compute_databox_type<typename intrp::Actions::InitializeInterpolator::
+                                   template return_tag_list<Metavariables>>;
+};
+
+template <typename PostHorizonFindCallback>
+struct MockMetavariables {
+  struct AhA {
+    using compute_items_on_source = tmpl::list<
+        ah::Tags::InverseSpatialMetricCompute<3, Frame::Inertial>,
+        ah::Tags::ExtrinsicCurvatureCompute<3, Frame::Inertial>,
+        ah::Tags::SpatialChristoffelSecondKindCompute<3, Frame::Inertial>>;
+    using vars_to_interpolate_to_target =
+        tmpl::list<gr::Tags::InverseSpatialMetric<3, Frame::Inertial>,
+                   gr::Tags::ExtrinsicCurvature<3, Frame::Inertial>,
+                   gr::Tags::SpatialChristoffelSecondKind<3, Frame::Inertial>>;
+    using compute_items_on_target = tmpl::list<>;
+    using compute_target_points =
+        intrp::Actions::ApparentHorizon<AhA, ::Frame::Inertial>;
+    using post_interpolation_callback =
+        intrp::callbacks::FindApparentHorizon<AhA>;
+    using post_horizon_find_callback = PostHorizonFindCallback;
+    // This `type` is so this tag can be used to read options.
+    using type = typename compute_target_points::options_type;
+  };
+  using interpolator_source_vars =
+      tmpl::list<gr::Tags::SpacetimeMetric<3, Frame::Inertial>,
+                 GeneralizedHarmonic::Pi<3, Frame::Inertial>,
+                 GeneralizedHarmonic::Phi<3, Frame::Inertial>>;
+  using interpolation_target_tags = tmpl::list<AhA>;
+  using temporal_id = Time;
```

Comment:
No, it doesn't need to be Time.  However, it needs to be something more than an arbitrary empty struct (i.e. it needs to be hashable, needs to have operator==, etc.), so making my own TemporalId struct doesn't help anything, especially since it will be used only in tests.  In a future PR (one that I have finished but haven't submitted because it depends on this and another PR) temporal_id will be changed to be a Tag whose type is what the current 'Time' is.